Assets/MRTK/SDK/Features/UX/Scripts/InteractiveElement/States/Definitions/CoreInteractionState.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License
+
+namespace Microsoft.MixedReality.Toolkit.UI.Interaction
+{
+    /// <summary>
+    /// A state with already defined state settting logic within an Interactive Element is considered a Core Interaction State.
+    /// </summary>
+    public enum CoreInteractionState
+    {
+        /// <summary>
```

Comment:
![image](https://user-images.githubusercontent.com/53493796/91920199-3277ca80-ec7d-11ea-86f1-249cbd6ed34d.png)

Reply:
Eye tracking and touch screen states can also be added to Core Interaction States if needed

Reply:
Couple questions:
1. Are the core states editable by developers / end users? 
2. I always found "focus", "touch" and "grab" to be a bit confusing, so have a few follow ups in case we can make these better. First, is focus trigger both near and far, or just far? I assume Touch triggers only when directly touching hologram, is that right? Finally, does Grab trigger both near and far? I would expect yes, but I think it might be only triggered when near, which is pretty confusing.

Reply:
1.  The state setting logic for a Core State is currently not editable.  The handlers are the entry points for state setting and a user would have to edit a handler method in BaseInteractiveElement.  But we could make a core state editable by having the state setting logic in a method that can be overwritten instead of directly in the handler.

2. Focus is both a near and far interaction state.  Touch is only a near interaction state, but the state setting logic for touch is not in this PR.  Grab is going to be an interesting state, because there is not a single entry point for state setting since we do not have a grab handler.  The state setting logic for grab is going to be based on the type of interaction.  For example, a grab can be registered if an object is being touched (near interaction), or it can be registered with just a click and drag if we are in VR (far interaction).

I was thinking of adding a button within each state to show a state description so it will be clear how the state is set.  

Also I could structure the state list with three sections: Near Interaction, Far Interaction, Near and Far Interaction. This way it is clear which states support a specific type of interaction.

Reply:
> The state setting logic for a Core State is currently not editable. The handlers are the entry points for state setting and a user would have to edit a handler method in BaseInteractiveElement. But we could make a core state editable by having the state setting logic in a method that can be overwritten instead of directly in the handler.

Can't a developer just call SetState(..) to set the state of an Interactable, from outside code?

Reply:
Ah yes they can, I thought you were requesting to edit the initial the state setting logic for a core state.  A developer can just call `interactiveElement.SetStateOn("Focus")` or `interactiveElement.SetStateOn(CoreInteractionState.Focus)`.  If they create their own state it will be the same: `interactiveElement.SetStateOn("MyNewState")`.

Reply:
> Focus is both a near and far interaction state. Touch is only a near interaction state, but the state setting logic for touch is not in this PR. Grab is going to be an interesting state, because there is not a single entry point for state setting since we do not have a grab handler. The state setting logic for grab is going to be based on the type of interaction. For example, a grab can be registered if an object is being touched (near interaction), or it can be registered with just a click and drag if we are in VR (far interaction).

This seems like something we could improve. I would still recommend we make these editable by end-users, and look to existing frameworks for common states. For example, [in CSS, buttons have three states: hover, focus, and active](https://zellwk.com/blog/style-hover-focus-active-states/).

For 3D buttons and interactables, we may need more states. Here are [oculus guidelines to consider](https://developer.oculus.com/learn/hands-design-interactions/), as well as [leap motion guidelines to consider](http://blog.leapmotion.com/design-playground-3d-user-interfaces/). For buttons:
Direct Interaction:
- Focus (hand near button)
- Contact (button physically touched by hand)
- Activated (button fully depressed)

Indirect interaction:
- Focus (this should be different from near focus, since visuals may need to be different)
- Activated (button pressed.)

For movable objects:
Near Interaction:
- Focus
- Moving

Far Interaction:
- Focus (should be different from near focus state, again visuals may need to be different)
- Moving

Note that for the focus states I mentioned that developers may need to provide different visual feedback for near focus vs. far focus. As long as there is a way to check in the interactive element whether the current interaction is a near or far interaction, the state could be combined. If not, you can just have a separate near focus and far focus state.

Reply:
Support for these states has been added to the to do list issue above